NST/src/main/java/com/ebay/nst/NSTServiceWrapperProcessor.java

```diff
@@ -299,9 +299,27 @@ protected void logResponseDetailsToConsole(NSTServiceWrapper<? extends NSTSchema
 			NSTHttpResponse response) {
 
 		if (response == null) {
+			Reporter.log("Response was null - skipping logging response details.", true);
 			return;
 		}
 
+		Map<String, String> responseHeaders = response.getHeaders();
+		if (responseHeaders != null && !responseHeaders.isEmpty()) {
+			Reporter.log("Response Headers:", true);
+			for (Map.Entry<String, String> entry : responseHeaders.entrySet()) {
+				Reporter.log(String.format("%s : %s", entry.getKey(), entry.getValue()), true);
+			}
+		} else {
+			Reporter.log("Response does not contain headers.", true);
+		}
+
+		if (response.getPayload() != null) {
+			Reporter.log("Response Payload:", true);
+			Reporter.log(response.getPayload(), true);
+		} else {
+			Reporter.log("Response does not contain a payload.", true);
+		}
```

Comment:
Indentation looks a little wonky here. Can you reformat? Could just be me

Reply:
Let's raise a separate PR for that. I don't want to lose sight of the changes made in this PR by reformatting the entire class.